alf/environments/simple/stochastic_with_risky_branch.py

```diff
@@ -0,0 +1,98 @@
+# Copyright (c) 2019 Horizon Robotics. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import gym
+from gym import spaces
+import numpy as np
+
+S0 = 0
+S1 = 1
+S2 = 2
+S3 = 3
+T = 4
+
+A0 = 0
+A1 = 1
+
+
+def _k(s, a):
+    return s * 10 + a
+
+
+class StochasticWithRiskyBranch(gym.Env):
+    """
+    A simple stochastic MDP
+    s0 -> a0 - 50%  -> s1 -> a0 -> T (reward=2)
+     |     |-- 50% -> s3 -> a1 -> T (reward=1)
+     |--> a1 - 100% -> s2 -> a0 -> T (reward=1.8)
+    All other actions terminates with reward 0.  T is the terminal state.
+
+    Optimal action at s0 is a1 with q(s0, a1) = 1.5.  Optimal q(s0, a0) = 1.8.
+
+    However, if it so happens that Q learning is conditioned on the action sequence, then
+    q(s0, a0, a0) will contain mostly experience of <s0, a0, s1, a0, s0> and very few <s0, a0, s3, a0, s0>,
+    leading to an average of about 100.
```

Comment:
100, 10, 80: these numbers should be updated?

Reply:
Good catch.